library/network/test/y2firewall/firewalld/service_reader_test.rb

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env rspec
+# encoding: utf-8
+#
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative "../../test_helper"
+require "y2firewall/firewalld"
+require "y2firewall/firewalld/service_reader"
+
+describe Y2Firewall::Firewalld::ServiceReader do
+  let(:firewalld) { Y2Firewall::Firewalld.instance }
+  let(:api) { instance_double("Y2Firewall::Firewalld::Api", state: "not_running") }
+
+  before do
+    allow(firewalld).to receive(:api).and_return(api)
+    `echo true`
```

Comment:
What is this line?

Reply:
I will add a comment, it was a workaround added for the definition of $CHILD_STATUS, in other case it will be nil at the moment of allowing calls over it.